ui/src/org/pentaho/di/ui/spoon/ExpandedContentManager.java

```diff
@@ -248,6 +253,32 @@ private static Spoon spoonInstance() {
   }
 
   /**
+   * isUnsupportedEnvironment
+   *
+   * @param parent Parent Transformation of the browser.
+   * @return ´true´ if the environment is unsupported ´false' otherwise.
+   */
+  private static boolean isUnsupportedEnvironment( TransGraph parent ) {
```

Comment:
don't see single responsibility for the method - it checks and in addition can show dialog and not consistent to the method name. Better to refactor this method.